x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/Verifier.java

```diff
@@ -51,4 +57,24 @@
 
         return failures;
     }
+
+    private static void checkForNonCollapsableSubselects(PhysicalPlan plan, List<Failure> failures) {
+        Holder<Boolean> hasLimit = new Holder<>(Boolean.FALSE);
+        Holder<List<Order>> orderBy = new Holder<>();
+        plan.forEachUp(p -> {
+            if (hasLimit.get() == false && p instanceof LimitExec) {
+                hasLimit.set(Boolean.TRUE);
+                return;
+            }
+            if (orderBy.get() == null && p instanceof OrderExec) {
+                orderBy.set(((OrderExec) p).order());
+                return;
+            }
+            if (hasLimit.get() && orderBy.get() != null && p instanceof OrderExec) {
+                if (((OrderExec) p).order().equals(orderBy.get()) == false) {
+                    failures.add(fail(p, "Cannot use ORDER BY on top of a subquery with ORDER BY and LIMIT"));
```

Comment:
Is the error message descriptive enough, tho? That _is_ actually possible[*], just not in any (i.e. "conflicting") configuration, right? Subselects are a gray area anyways, so I guess this could also remain like this, but alternatively we could add some notes in the limitations docs or somehow rephrase this message -- up to you.

[*] Didn't run it on the new code, but I guess this should work: `SELECT * FROM (SELECT * FROM test_emp ORDER BY emp_no LIMIT 10) ORDER BY emp_no LIMIT 5`

Reply:
Yep it passes, since it's the exact same order By so it can be executed once and then merge the limits to 5: https://github.com/elastic/elasticsearch/pull/72991/files#diff-147ac28e10bf6b2342a276203e3b250ccd593163c3aa087ee6d72ed19572c163R147

Reply:
Regarding the error message, imho we shouldn't add more details, about not supporting an ORDER BY on top of the result set returned internally, I think it will just complicate things, and regarding the limitations we already state about supporting queries that can be "flattened".